tools/engine_tool/lib/src/commands/test_command.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:path/path.dart' as p;
+
+import '../build_utils.dart';
+import '../gn_utils.dart';
+import '../proc_utils.dart';
+import '../worker_pool.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'test' command.
+final class TestCommand extends CommandBase {
+  /// Constructs the 'build' command.
+  TestCommand({
+    required super.environment,
+    required Map<String, BuilderConfig> configs,
+  }) {
+    builds = runnableBuilds(environment, configs);
+    debugCheckBuilds(builds);
+    argParser.addOption(
+      configFlag,
+      abbr: 'c',
+      defaultsTo: 'host_debug',
+      help: 'Specify the build config to use',
+      allowed: <String>[
+        for (final Build config in runnableBuilds(environment, configs))
+          config.name,
+      ],
+      allowedHelp: <String, String>{
+        for (final Build config in runnableBuilds(environment, configs))
+          config.name: config.gn.join(' '),
+      },
+    );
+  }
+
+  /// List of compatible builds.
+  late final List<Build> builds;
+
+  @override
+  String get name => 'test';
+
+  @override
+  String get description => 'Runs a test target';
+
+  @override
+  Future<int> run() async {
+    final String configName = argResults![configFlag] as String;
+    final Build? build =
+        builds.where((Build build) => build.name == configName).firstOrNull;
+    if (build == null) {
+      environment.logger.error('Could not find config $configName');
+      return 1;
+    }
+
+    final Map<String, TestTarget> allTargets = await findTestTargets(
```

Comment:
The test also needs to run `gclient sync -D` if dependencies have changed. I'd add a TODO here for me :)

I switched from an old engine branch to this branch, ran `et test`, and got this error:

```
PS C:\Code\f\engine\src\flutter> .\bin\et test
[2024-03-22 09:20:13.051926] ERROR: Process "gn ls C:\Code\f\engine\src\out\host_debug --type=executable --testonly=true --as=label" failed exitCode=1
STDOUT:
ERROR at //flutter/skia/BUILD.gn:28:7: Undefined identifier
  if (skia_enable_flutter_defines) {
      ^--------------------------
See //flutter/tools/path_ops/BUILD.gn:14:12: which caused the file to be included.
  deps = [ "//flutter/skia" ]
           ^---------------
//out/host_debug/environment.x64
HOMEDRIVE=C:STDERR:

[2024-03-22 09:20:13.065926] ERROR: FatalError caught in main. Please file a bug
error: Process "gn ls C:\Code\f\engine\src\out\host_debug --type=executable --testonly=true --as=label" failed exitCode=1
STDOUT:
ERROR at //flutter/skia/BUILD.gn:28:7: Undefined identifier
  if (skia_enable_flutter_defines) {
      ^--------------------------
See //flutter/tools/path_ops/BUILD.gn:14:12: which caused the file to be included.
  deps = [ "//flutter/skia" ]
           ^---------------
//out/host_debug/environment.x64
HOMEDRIVE=C:STDERR:


stack: #0      Logger.fatal (package:engine_tool/src/logger.dart:123:5)
#1      fatalIfFailed (package:engine_tool/src/proc_utils.dart:234:22)
#2      findTestTargets (package:engine_tool/src/gn_utils.dart:45:3)
<asynchronous suspension>
#3      TestCommand.run (package:engine_tool/src/commands/test_command.dart:61:48)
<asynchronous suspension>
#4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#5      ToolCommandRunner.run (package:engine_tool/src/commands/command_runner.dart:65:14)
<asynchronous suspension>
#6      main (package:engine_tool/main.dart:69:19)
<asynchronous suspension>
```

Reply:
Whoops, never mind, you moved the fetch TODO comment into `runBuild`.